### PR TITLE
fix(docs-infra): amend color of code links inside a tags

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
when some auto code links fail to happen they are added manually
with the md ``[`code text`](link)``, these generate anchor elements which
contain a code element, such code element does not get the correct text
color, this commit fixes such issue

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [x] docs style change

## What is the current behavior?

Issue Number: N/A

Code links automatically generated by the docs pre-processor get a light blue color, but those added manually in md files don't

For example:
Screenshot at 2021-09-25 13-12-39

## What is the new behavior?

Code links manually added in md files get the blue color as the other auto-generated links:

For example:
Screenshot at 2021-09-25 13-13-02

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by dario-piotrowicz** [comment]
### Note
Unfortunately this does not make the manual links 100% the same as auto-generates as in functions the brackets gets also colored/included in the link, which is not what happens with the auto-generated links:
Screenshot at 2021-09-25 13-34-25 vs
Screenshot at 2021-09-25 13-37-14

This is unfortunate but I don't think there is much that can be done here about it as that is related to the docs generation

**Review by petebacondarwin — APPROVED** [review]
Thanks @dario-piotrowicz.

I agree that this does not make things worse. And in the bigger picture, I am not worried about the difference TBH.

By the way, you can use double backticks to mark inline code that will contain backticks to make your git message cleaner:

```
``[`code text`](link)``
```

**Comment by dario-piotrowicz** [comment]
> Thanks @dario-piotrowicz.
> 
> I agree that this does not make things worse. And in the bigger picture, I am not worried about the difference TBH.
> 
> By the way, you can use double backticks to mark inline code that will contain backticks to make your git message cleaner:
> 
> ```
> ``[`code text`](link)``
> ```

Thanks, I've updated the commit message with the double backticks :slightly_smiling_face::+1: 

(about the difference with the brackets... yeah it's a bit of an unimportant detail, I am sure there are much more relevant things than that :sweat_smile:)

File: aio/src/styles/2-modules/code/_code-theme.scss [hunk 1]
```diff
@@ -87,17 +87,22 @@
...
     code {
       a {
-        color: if($is-dark-theme, constants.$cyan, constants.$darkblue);
+        color: $link-color;
       }
     }
 
     :not(h1):not(h2):not(h3):not(h4):not(h5):not(h6):not(pre) {
-      > code {
+      & > code {
```
Comment (gkalpak) [inline aio/src/styles/2-modules/code/_code-theme.scss:98]: OOC, why this change?

File: aio/src/styles/2-modules/code/_code-theme.scss [hunk 2]
```diff
@@ -87,17 +87,22 @@
...
         background-color: if($is-dark-theme, constants.$darkgray, rgba(constants.$lightgray, 0.3));
         color: if($is-dark-theme, constants.$white, constants.$darkgray);
       }
+
+      &:is(a) > code {
```
Comment (gkalpak) [inline aio/src/styles/2-modules/code/_code-theme.scss:103]: I would rather not use `:is()` yet (at least until we switch to v13 and drop IE - and also Safari has is not 100% spec-compliant).

How about:

```diff
   .sidenav-content {
     code {
       a {
         color: if($is-dark-theme, constants.$cyan, constants.$darkblue);
       }
     }
 
     :not(h1):not(h2):not(h3):not(h4):not(h5):not(h6):not(pre) {
       > code {
         background-color: if($is-dark-theme, constants.$darkgray, rgba(constants.$lightgray, 0.3));
-        color: if($is-dark-theme, constants.$white, constants.$darkgray);
+
+        &:not(a) {
+          color: if($is-dark-theme, constants.$white, constants.$darkgray);
+        }
       }
     }
   }
```

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx, @dario-piotrowicz ❤️
One suggestion to avoid `:is()` and also avoid style duplication. Otherwise lgtm ✨

File: aio/src/styles/2-modules/code/_code-theme.scss (hunk 1 above)
Comment (dario-piotrowicz) [reply aio/src/styles/2-modules/code/_code-theme.scss:98]: ... personal preference :see_no_evil: :sweat_smile: :disappointed:

File: aio/src/styles/2-modules/code/_code-theme.scss (hunk 2 above)
Comment (dario-piotrowicz) [reply aio/src/styles/2-modules/code/_code-theme.scss:103]: yes totally agree with avoiding the `:is` I did it on my other PR but forgot to do it here :sweat: 

yeah I think your change should definitely be the right solution, I will test it out this evening and commit it :slightly_smiling_face::+1:

Thanks a lot :smiley: 

(and sorry yeah....my solution was pretty ugly wasn't it :sweat_smile:)

Comment (dario-piotrowicz) [reply aio/src/styles/2-modules/code/_code-theme.scss]: Hi @gkalpak sorry I could not go with your solution as it didn't work :sweat: 

the issue is that your `&:not(a)` isn't what we need, `a` is the parent and `code` is the element we want to target :sweat: 

So I went with this: https://github.com/angular/angular/pull/43586/commits/3ff1f3fa2bca9bf528bfbddccf6dd37f81872cf7

Please have a look it's not as clean but it removes both the duplication and the `:is` :slightly_smiling_face:

File: aio/src/styles/2-modules/code/_code-theme.scss (hunk 1 above)
Comment (dario-piotrowicz) [reply aio/src/styles/2-modules/code/_code-theme.scss:96]: is this ok? :sweat_smile: :sweat: :see_no_evil:

File: aio/src/styles/2-modules/code/_code-theme.scss (hunk 2 above)
Comment (gkalpak) [reply aio/src/styles/2-modules/code/_code-theme.scss]: Oh, you're right 🤦‍♂️
Yeah, I meant, `$:not(a) > code` (you got the spirit, though 😛).

File: aio/src/styles/2-modules/code/_code-theme.scss (hunk 1 above)
Comment (gkalpak) [reply aio/src/styles/2-modules/code/_code-theme.scss:96]: I am not a big fan (and it is generally better to remain consistent), but I am not going to block this PR over this :wink:

File: aio/src/styles/2-modules/code/_code-theme.scss [hunk 3]
```diff
@@ -87,21 +87,17 @@
...
 
     :not(h1):not(h2):not(h3):not(h4):not(h5):not(h6):not(pre) {
       & > code {
         background-color: if($is-dark-theme, constants.$darkgray, rgba(constants.$lightgray, 0.3));
-        color: if($is-dark-theme, constants.$white, constants.$darkgray);
       }
-
-      &:is(a) > code {
-        color: $link-color;
+      &:not(a) > code {
```
Comment (gkalpak) [inline aio/src/styles/2-modules/code/_code-theme.scss:99]:
```suggestion

      &:not(a) > code {
```

😱

**Review by gkalpak — APPROVED** [review]
Thx, @dario-piotrowicz! (One minor nit - otherwise good to go 🚀)

File: aio/src/styles/2-modules/code/_code-theme.scss (hunk 3 above)
Comment (dario-piotrowicz) [reply aio/src/styles/2-modules/code/_code-theme.scss:99]: sorry :sweat:

File: aio/src/styles/2-modules/code/_code-theme.scss (hunk 1 above)
Comment (dario-piotrowicz) [reply aio/src/styles/2-modules/code/_code-theme.scss:96]: I imagined that `& >` was actually more consistent (I didn't really look into it :sweat:)

if you prefer the other version I will just revert it sorry :slightly_smiling_face::+1:

**Review by gkalpak — APPROVED** [review]

**Comment by gkalpak** [comment]
Thx, @dario-piotrowicz! This should be good to go as soon as CI is green. (It seems that a rebase is needed currently.)

**Comment by dario-piotrowicz** [comment]
> Thx, @dario-piotrowicz! This should be good to go as soon as CI is green. (It seems that a rebase is needed currently.)

Actually this should be up to date already :confused:

**Comment by gkalpak** [comment]
> Actually this should be up to date already 😕

Well, CI [strongly disagrees][1] :grin:
I'll rebase this.

[1]: https://circleci.com/gh/angular/angular/1061241

**Comment by dario-piotrowicz** [comment]
> > Actually this should be up to date already confused
> 
> Well, CI [strongly disagrees](https://circleci.com/gh/angular/angular/1061241) grin I'll rebase this.

Sorry.... I did check it and the branch seemed up to date.... sorry :sweat:

**Comment by gkalpak** [comment]
No worries :smiley:

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 2a58ff3e416e2cfdec0e670489f9ebe5baaf2e22.
